lib/common_fw/common_fw.cpp

```diff
@@ -77,30 +77,32 @@ uint8_t make_packet(uint8_t id, bool error) {
  * Populated the fields of the valve and returns the action to be taken
  * This is a pretty beefy function; can we split this up
  */
-int8_t decode_received_packet(uint8_t *packet, valveInfo *valve, valveInfo valves[], int numValves, int DEBUG) {
+int8_t decode_received_packet(uint8_t *_command, valveInfo *valve, valveInfo valves[], int numValves) {
   uint8_t i = 0;
-  uint16_t command_open = (packet[i++] << 8) & packet[i++];
+  uint16_t command_open = (_command[i++] << 8) & _command[i++];
   if (command_open != PACKET_START) {
     return -1;
   }
-  int valve_id = packet[i++];
+  int valve_id = _command[i++];
 
-  int action = packet[i++]; // currently, action is assumed to be a single int.
-  uint16_t checksum = (packet[i++] << 8) & packet[i++];
-  uint16_t _check = Fletcher16(packet+2, 2); // just id and action both of which are ints.
+  int action = _command[i++]; // currently, action is assumed to be a single int.
+  uint16_t checksum = (_command[i++] << 8) & _command[i++];
+  uint16_t _check = Fletcher16(_command+2, 2); // just id and action both of which are ints.
 
-  debug(checksum, DEBUG);
-  debug(_check, DEBUG);
+  int size = sprintf(packet,"%x",checksum);
+  debug(packet, size);
+  size = sprintf(packet,"%x", _check);
+  debug(packet, size);
```

Comment:
Do you mean `_command` instead of `packet` here?

Reply:
no I'm using packet as a temp variable (char *) to hold the result string representation of the checksum, which will then get printed out.